### PR TITLE
Handle errors in team email invites

### PR DESCRIPTION
**Pull request by buoyad**
Pipes through an error message to desktop & mobile views. Also does some cleanup of the components (de-recomposed desktop view, etc). On desktop the input is replaced by the malformed emails. The mobile error can show up if you have a contact with an invalid email address. r? @keybase/react-hackers

### CODE REVIEW

File: shared/teams/invite-by-email/index.desktop.js [hunk 1]
```diff
@@ -1,34 +1,22 @@
 // @flow
 import * as React from 'react'
-import {Box, Button, ClickableBox, Dropdown, Input, PopupDialog, Text, ButtonBar} from '../../common-adapters'
-import {globalStyles, globalMargins, globalColors, isMobile} from '../../styles'
+import {Set} from 'immutable'
```
Comment (chrisnojima) [inline shared/teams/invite-by-email/index.desktop.js]: lets always import immutable as I

File: shared/teams/invite-by-email/index.desktop.js [hunk 2]
```diff
@@ -46,81 +34,115 @@ const _makeDropdownItem = (item: string) => (
...
-            inputStyle={styleInput}
-          />
-        </Box>
+            >
+              <Text style={{margin: globalMargins.tiny}} type="Body">
+                Add these team members to {props.name} as:
+              </Text>
+              <ClickableBox
+                onClick={() => props.onOpenRolePicker(this.state.role, this._setRole)}
+                underlayColor="rgba(0, 0, 0, 0)"
```
Comment (chrisnojima) [inline shared/teams/invite-by-email/index.desktop.js]: globalColors.transparent

File: shared/teams/invite-by-email/index.js.flow [hunk 3]
```diff
@@ -1,5 +1,6 @@
 // @flow
 import * as React from 'react'
+import {type Set} from 'immutable'
```
Comment (chrisnojima) [inline shared/teams/invite-by-email/index.js.flow]: same here

**Comment by buoyad** [comment]
@chrisnojima thanks, addressed comments, PTAL

**Pull request merged**